app/models/partnership_csv_upload.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class PartnershipCsvUpload < ApplicationRecord
+  has_one_attached :csv
+  # does this need a delivery partner reference ?
+  belongs_to :lead_provider, optional: true
+
+  validate :csv_validation
+
+  MAX_FILE_SIZE = 2.megabytes
+
+private
+
+  def csv_validation
+    return unless csv.attached?
+
+    if csv.content_type != "text/csv"
```

Comment:
Excitingly, the file you use in tests fails... It has type `application/vnd.ms-excel`

Reply:
Uploading a CSV still does not work. It might be a windows thing? Whatever it is, I expect it will affect our users

Reply:
https://stackoverflow.com/questions/1201945/how-is-mime-type-of-an-uploaded-file-determined-by-browser#:~:text=when%20uploading%20a%20csv%20file%20from%20a%20windows%20system%20with%20microsoft%20excel%20installed%2C%20chrome%20will%20report%20this%20as

Reply:
I think we'd be better off detecting the type of uploaded file, rather than taking the browser's word for it